tidb-in-kubernetes/tkctl.md

```diff
@@ -0,0 +1,387 @@
+local-pv-ed2ffe50   tikv-demo-cluster-tikv-2   Bound    1476Gi     172.16.4.157   /mnt/disks/local-pv13
+local-pv-74ee0364   pd-demo-cluster-pd-0       Bound    1476Gi     172.16.4.155   /mnt/disks/local-pv46
+local-pv-842034e6   pd-demo-cluster-pd-1       Bound    1476Gi     172.16.4.158   /mnt/disks/local-pv74
+local-pv-e54c122a   pd-demo-cluster-pd-2       Bound    1476Gi     172.16.4.156   /mnt/disks/local-pv72
+```
+
+### tkctl debug [pod_name]
+
+该命令用于诊断 TiDB 集群中的 Pod。
+
+实际使用时，该命令会在目标 Pod 的宿主机上以指定镜像启动一个 debug 容器，该容器会与目标 Pod 中的容器共享 namespace，因此可以无缝使用 debug 容器中的各种工具对目标容器进行诊断。
+
+| 参数            | 缩写      | 描述                                                                                                                            |
+| -----           | --------- | -----------                                                                                                                     |
+| --image         |           | 指定 debug 容器使用的镜像，默认为 `pingcap/tidb-debug:lastest`                                                                  |
+| --container     | -c        | 选择需要诊断的容器，默认为 Pod 定义中的第一个容器                                                                               |
+| --docker-socket |           | 指定目标节点上的 Docker Socket，默认为 `/var/run/docker.sock`                                                                   |
+| --privileged    |           | 是否为 debug 容器开启 [privileged](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) 模式 |
+
+> **注意：**
+>
+> Debug 容器使用的默认镜像包含了绝大多数的诊断工具，因此体积较大，假如只需要 `pd-ctl` 和 `tidb-ctl`，可以使用 `--image=pingcap/tidb-control:latest` 来指定使用 `tidb-control` 镜像。
+
+示例如下：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tkctl debug demo-cluster-tikv-0
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+ps -ef
+```
+
+由于 debug 容器和目标容器拥有不同的根文件系统，在 `tidb-debug` 容器中使用 GDB 和 perf 等工具时可能会碰到一些问题，下面将补充说明如何解决这些问题。
+
+#### GDB
+
+使用 GDB 调试目标容器中的进程时，需要将 `program` 参数设置为目标容器中的可执行文件。假如是在 `tidb-debug` 以外的其它 debug 容器中进行调试，或者调试的目标进行 pid 不为 1，则需要使用 `set sysroot` 命令调整动态链接库的加载位置。操作如下：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tkctl debug demo-cluster-tikv-0
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+gdb /proc/${pid:-1}/root/tikv-server 1
+```
+
+`tidb-debug` 中预配置的 `.gdbinit` 会将 `sysroot` 设置为 `/proc/1/root/`，因此在 `tidb-debug` 中，假如目标容器的 pid 为 1，则下面的命令可以省略。
+
+{{< copyable "shell-regular" >}}
+
+```shell
+(gdb) set sysroot /proc/${pid}/root/
+```
+
+开始调试：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+(gdb) thread apply all bt
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+(gdb) info threads
+```
+
+#### Perf（以及火焰图）
+
+使用 `perf` 命令和 `run-flamegraph.sh` 脚本时，需要将目标容器的可执行文件拷贝到 Debug 容器中：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tkctl debug demo-cluster-tikv-0
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+cp /proc/1/root/tikv-server /
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+./run_flamegraph.sh 1
```

Comment:
This command will generate an SVG format graph and upload to https://transfer.sh, so users can view or download it from the output URL.
I think this should be noted here.